### PR TITLE
implement service as gateway

### PR DESCRIPTION
**Pull request by veidl**

### CODE REVIEW

File: src/main/java/at/ac/fhcampuswien/car_rental/config/CustomGlobalFilter.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import org.springframework.cloud.gateway.filter.GlobalFilter;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.security.core.context.ReactiveSecurityContextHolder;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ServerWebExchange;
+import reactor.core.publisher.Mono;
+
+@Component
+public class CustomGlobalFilter implements GlobalFilter {
```
Comment (mashkie) [inline src/main/java/at/ac/fhcampuswien/car_rental/config/CustomGlobalFilter.java:13]: The GlobalFilter is from spring gateway and the Mono<> is a type which all gateway related methods should be of?

**Review by mashkie — APPROVED** [review]

File: src/main/java/at/ac/fhcampuswien/car_rental/config/CustomGlobalFilter.java (hunk 1 above)
Comment (veidl) [reply src/main/java/at/ac/fhcampuswien/car_rental/config/CustomGlobalFilter.java:13]: Yes spring gateway is build upon the reactor framework. The global filter is used for each request that qualifies for a predicate in the RouteLocator.

Mono's (or Fluxes) are non-blocking reactive streams.
Find more info on the official project description: https://github.com/reactor/reactor-core

**Comment by Henchway** [comment]
Generally, is the latest version of the car_rental_service deployed to release/part1? @SahilSinghStudierender 
Otherwise you shouldn't merge this yet and adapt the workflow to deploy to release/part2, also some other details in the workflow file still need to be adapted.

**Comment by Henchway** [comment]
Misclicked :|

**Review by SahilSinghStudierender — APPROVED** [review]

**Pull request merged**